### PR TITLE
Deploy dmg and deb files built on Travis to downloads.mixxx.org.

### PR DESCRIPTION
**Pull request by rryan**
Adds an RSA key that is authorized to login to downloads-hostgator.mixxx.org. This key is encrypted with a password which is provided as a Travis secure variable.

Builds for branches are uploaded to http://downloads.mixxx.org/builds/travis/branch-name/.

PR builds cannot be deployed this way because they would require the key to be decrypted (and PRs do not have access to secure variables to prevent easy exfiltration of the key).

TESTED:
- https://travis-ci.org/github/mixxxdj/mixxx/builds/710054843
- http://downloads.mixxx.org/builds/travis/travis-deploy/

### CODE REVIEW

File: build/travis/deploy.sh [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+USER=mixxx
+HOSTNAME=downloads-hostgator.mixxx.org
+TRAVIS_DESTDIR=public_html/downloads/builds/travis
+SSH_KEY=../build/certificates/downloads-hostgator.mixxx.org.key
+SSH="ssh -i ${SSH_KEY} -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null"
+
+if [[ "${TRAVIS_PULL_REQUEST}" != "false" ]]; then
+  echo "Skipping deploy since we are in a pull request."
+  exit 0
+fi
```
Comment (Be-ing) [inline build/travis/deploy.sh:17]: Can we deploy unsigned builds for PRs? This would help a lot with getting people to test PRs.

Comment (Be-ing) [reply build/travis/deploy.sh:17]: Oh, nevermind. This script is just deploying, not signing.

Comment (Be-ing) [reply build/travis/deploy.sh:17]: I don't think this check in the script is necessary:

> Note that pull request builds skip the deployment step altogether.

https://docs.travis-ci.com/user/deployment/#pull-requests

**Comment by rryan** [comment]
Travis is only failing due to timeouts on the macOS scons build. This should be ready to go.

Open TODOs that I think can be done in separate PRs:
* Improve the filenames cpack produces to include a "gitXXXX" user-friendly version, the Git hash, the branch name, etc.
* Write a "latest" symlink to the upload directory.
* Change the upload directory to not be Travis-specific once we are happy with the builds it produces. (AppVeyor and Travis should probably publish to the same place).

**Pull request merged**